### ISSUE TITLE
ALB listener .addTargets bug with cdk.Duration and .toString (v0.36.0)

### ISSUE THREAD

**Issue by joshlartz** [aws/aws-cdk#3080]
* **I'm submitting a ...**
  - [X] :beetle: bug report

* **What is the current behavior?**
*If the current behavior is a :beetle:bug:beetle:: Please provide the steps to reproduce*

when calling .addTargets() against an elbv2.ApplicationListener with a config of:
{ ...,
  deregistrationDelay: cdk.Duration.seconds(5),
  ...
}
the method tries to call .toString() on the Duration object and fails with an error about using .toSecond, etc. instead.

https://github.com/awslabs/aws-cdk/blob/master/packages/%40aws-cdk/aws-elasticloadbalancingv2/lib/shared/base-target-group.ts#L208

* **What is the expected behavior (or behavior of feature suggested)?**

.toString() should not be called against a Duration object. It results in a runtime error. 

* **Please tell us about your environment:**
  
  - CDK CLI Version: 0.36.0
  - Module Version: 0.36.0
  - OS: OSX Mojave
  - Language: all

* **Other information** (e.g. detailed explanation, stacktraces, related issues, suggestions how to fix, links for us to have context, eg. associated pull-request, stackoverflow, gitter, etc)

**Comment by mhuebner** [comment]
Same here:

```
rror: Argument to Intrinsic must be a plain value object, got () => {
            throw new Error(`Duration.toString() was used, but .toSeconds, .toMinutes or .toDays should have been called instead`);
        }
```

**Comment by rvdwijngaard** [comment]
+1

**Comment by nmussy** [comment]
Should be fixed with #3075
